compliance_checker/cf/cf.py

```diff
@@ -2899,77 +2900,84 @@ def check_climatological_statistics(self, ds):
                     'mode',
                     'median']
 
-        # first, to determine whether or not we have a climatological time
-        # variable, we need to make sure it has the attribute "climatology",
-        # but not the attribute "bounds"
-        meth_regex = "(?:{})".format("|".join(methods))
-
-        # find the variables with climatology attributes
-        clim_containing_vars = ds.get_variables_by_attributes(
+        # find any climatology axies variables; any variables which contain climatological stats will use
+        # these variables as coordinates
+        clim_time_coord_vars = ds.get_variables_by_attributes(
             climatology=lambda s: s is not None)
 
-        for clim_var in clim_containing_vars:
-            if hasattr(clim_var, 'bounds'):
-                reasoning.append('Variable {} has a climatology attribute and cannot also have a bounds attribute.'.format(clim_var.name))
+        # first, to determine whether or not we have a valid climatological time
+        # coordindate variable, we need to make sure it has the attribute "climatology",
+        # but not the attribute "bounds"
+        for clim_coord_var in clim_time_coord_vars:
+            if hasattr(clim_coord_var, 'bounds'):
+                reasoning.append('Variable {} has a climatology attribute and cannot also have a bounds attribute.'.format(clim_coord_var.name))
                 result = Result(BaseCheck.MEDIUM,
                                 False,
                                 (self.section_titles['7.4']),
                                 reasoning)
                 ret_val.append(result)
                 return ret_val
+
             # make sure the climatology variable referenced actually exists
-            elif clim_var.climatology not in ds.variables:
+            elif clim_coord_var.climatology not in ds.variables:
                 reasoning.append("Variable {} referenced in time's climatology attribute does not exist".format(ds.variables['time'].climatology))
                 result = Result(BaseCheck.MEDIUM,
                                 False,
                                 (self.section_titles['7.4']),
                                 reasoning)
                 ret_val.append(result)
                 return ret_val
+
             # handle 1-d and 2d coordinate bounds
```

Comment:
I can't imagine any case where the time coordinate would be 2-dimensional??
As far as I can tell `clim_coord_var.ndim` should always be 1, which would allow the code and reasoning message below to be much simpler...
(I realise this code is from a couple of years ago ... https://github.com/ioos/compliance-checker/commit/1f7c70b413598a078863b9f11141c5a8d6f6bdad)

Reply:
http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#climatological-statistics

> Intervals of climatological time are conceptually different from ordinary time intervals; a given interval of climatological time represents a set of subintervals which are not necessarily contiguous. To indicate this difference, a climatological time coordinate variable does not have a bounds attribute. Instead, it has a climatology attribute, which names a variable with dimensions (n,2), n being the dimension of the climatological time axis.

So yeah, this probably could be simplified.